ACPs/176-dynamic-evm-gas-limit-and-price-discovery-updates/README.md

```diff
@@ -0,0 +1,138 @@
+
+Note that the values of $T$, $R$, and $C$ are updated **after** the execution of block $b$, which means they only take effect in determining the gas price of block $b+1$. The change to each of these values in block $b$ does not effect the gas price for transaction includedin block $b$ itself.
+
+Allowing block builders to adjust the target gas consumption rate in blocks that they makes it such that the effect target gas consumption rate should loosely converge over time around the stake weighted average value set by validators of the network. This is because the number of blocks each validator produces is propotional to their stake weight. This means that an individual validator's effect on the resulting target gas consumption for the network is propotional to their stake weight.
+
+Currently, it is not valid for a block to contain zero transactions since it would not have any effect and would be a waste of resources to accept into the blockchain. However, with this change, it is possible for blocks with zero transactions to still effect value of $T$. To ensure that validators are able to help influence the value of $T$ even if there are no transactions to be included at the time they are proposing a block, blocks with no transactions that alter the value of $T$ will now be considered valid. This makes it such that validators would not need to create and include a no-op transaction just be able to produce a block to alter the current value of $T$ if there are no pending transactions when it is their turn to propose a block.
+
+### Configuration Parameters
+As noted above, the gas price determination mechanism depends on the values of $T$, $M$, $K$, $C$, and $R$ to be set as parameters. $T$ is adjusted dynamically from its initial value based on $D$ and $P$, and the values of $R$ and $C$ are derived from $T$. 
+
+Parameters at activation on the C-Chain are:
+
+<div align="center">
+
+| Parameter | Description | P-Chain Configuration|
+| - | - | - |
+| $T$ | initial target gas consumed per second | $1,500,000$ |
+| $M$ | minimum gas price | $1*10^{-18}$ AVAX  |
+| $K$ | gas price update constant | $2,164,043$ |
+| $D$ | target change bound divisor | $1024$ |
+| $P$ | minimum target gas consumption per second | $1,000,000$
+
+</div>
+
+$T$ was chosen as the current target gas consumption rate on the C-Chain. Avalanche L1s may change this value to match their current gas consumption rate if they would like.
+
+$M$ was chosen as the minimum possible denomination of the native EVM asset, such that the gas price will be more likely to consistently be in a range of price discovery. The price discovery mechanism has already been battle tested on the P-Chain (and prior to that on Ethereum for blob gas prices as defined by EIP-4844), giving confidence that it will correct react to any increase in network usage in order to prevent a DOS attack.
```

Comment:
```suggestion
$M$ was chosen as the minimum possible denomination of the native EVM asset, such that the gas price will be more likely to consistently be in a range of price discovery. The price discovery mechanism has already been battle tested on the P-Chain (and prior to that on Ethereum for blob gas prices as defined by EIP-4844), giving confidence that it will correctly react to any increase in network usage in order to prevent a DOS attack.
```